AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -66,7 +66,9 @@ public final class APIClient: APIClientProtocol {
 
         log(urlRequest: urlRequest, request: request)
         
-        requestCounter += 1
+        DispatchQueue.main.async {
```

Comment:
Does it have to be `main` ?
We can do a custom dispatch queue

Reply:
I think main is the best choice, no need for another queue I would say.